apis/common/v1/resource.go

```diff
@@ -226,6 +226,7 @@ type ResourceSpec struct {
 // ResourceStatus represents the observed state of a managed resource.
 type ResourceStatus struct {
 	ConditionedStatus `json:",inline"`
+	ObservedStatus    `json:",inline"`
```

Comment:
Should XRs and claims have `observedGeneration` too? I think that would be ideal.

Unfortunately they work a little differently, we'd need to update `internal/xcrd` in c/c.

Reply:
Edit: Nevermind, this PR already indicates that that's the goal.